data/data.go

```diff
@@ -86,52 +42,86 @@ func (e Event) HumanTime() string {
 	return time.Unix(e.Time/1000, 0).Format(time.RFC1123)
 }
 
-func NewMeetupSchedule() *MeetupSchedule {
-	return &MeetupSchedule{
-		events: make(map[string][]Event),
-	}
-}
+type Schedule struct {
+	key string
 
-type MeetupSchedule struct {
-	events map[string][]Event
+	Label  string
+	Events []Event
 }
 
-func (m *MeetupSchedule) Add(name string, events []Event) {
-	m.events[name] = events
-}
+type Schedules []*Schedule
+
+func (s *Schedule) FetchEvents(client Client) (err error) {
+	url := fmt.Sprintf("https://api.meetup.com/%s/events?status=upcoming", s.key)
+
+	data, err := client.Get(url)
 
-func (m *MeetupSchedule) BoulderEvents() []Event {
-	return nextThree(m.events["Boulder-Gophers"])
+	if err != nil {
+		return err
+	}
+
+	err = json.Unmarshal(data, &s.Events)
+
+	if err != nil {
+		return err
+	}
+
+	sort.SliceStable(s.Events, func(i, j int) bool {
+		return s.Events[i].Time < s.Events[j].Time
+	})
+
+	return err
 }
 
-func (m *MeetupSchedule) DenverEvents() []Event {
-	return nextThree(m.events["Denver-Go-Language-User-Group"])
+func (s *Schedule) Next(count int) []Event {
```

Comment:
So I *think* a solution here would be to not modify `s.Events` inside of `FetchEvents` and move more closely to the original implementation.  I really need to check out those screencasts you did to get a better understanding of the context when looking at this code. 

This may be something best suited for reviewing during a pairing session.